test/fs/test_fs_js_api.c

```diff
@@ -4,6 +4,43 @@
 #include <assert.h>
 #include <fcntl.h>
 
+void test_fs_allocate() {
+    EM_ASM(
+        FS.writeFile("allocatetestfile", 'a=1\nb=2\n');
+    );
+    struct stat allocateStat;
+    stat("allocatetestfile", &allocateStat);
+    assert(allocateStat.st_size == 8);
+
+    EM_ASM(
```

Comment:
Perhaps a comment for this subsection, "allocate more space at the very end".

Later parts could be "allocate from the start", "middle" etc.